executor/builder.go

```diff
@@ -2346,7 +2349,10 @@ func (b *executorBuilder) buildIndexLookUpMergeJoin(v *plannercore.PhysicalIndex
 	innerPlan := v.Children()[v.InnerChildIdx]
 	innerTypes := make([]*types.FieldType, innerPlan.Schema().Len())
 	for i, col := range innerPlan.Schema().Columns {
-		innerTypes[i] = col.RetType
+		innerTypes[i] = col.RetType.Clone()
+		if innerTypes[i].EvalType() == types.ETString {
```

Comment:
Could we only modify the `flen` for join keys?

Reply:
Only change the `flen` for join keys will make the codes more complicated to be understand. It should be ok since the `ConvertTo` only happens when build hash map and construct lookup keys.